chamge/amqp-edge-backend.c

```diff
@@ -47,8 +68,9 @@ _amqp_rpc_login (amqp_connection_state_t amqp_conn, amqp_socket_t * amqp_socket,
   g_return_val_if_fail (connection_info.user != NULL, CHAMGE_RETURN_FAIL);
   g_return_val_if_fail (connection_info.password != NULL, CHAMGE_RETURN_FAIL);
 
-  g_return_val_if_fail (amqp_socket_open (amqp_socket,
-          connection_info.host, connection_info.port) >= 0, CHAMGE_RETURN_FAIL);
+  RETURN_FAIL_IF_TRUE (amqp_socket_open (amqp_socket,
+          connection_info.host, connection_info.port) < 0,
+      "error in socket open");
```

Comment:
Restrictly speaking, assertion isn't recommended to check a return value. 
In addition, this pattern isn't good when tracing stack in debugger like gdb or lldb.

The below pattern is better to catch backtrace.
```
if (amqp_socket_open (...) < 0) {
  // add log
  goto failed;
}

```